crates/storage/store_db/in_memory.rs

```diff
@@ -47,6 +47,8 @@ pub struct StoreInner {
     invalid_ancestors: HashMap<BlockHash, BlockHash>,
     // Stores current Snap State
     snap_state: SnapState,
+    // Stores fetched headers during a fullsync
+    fullsync_headers: HashMap<BlockNumber, BlockHeader>,
```

Comment:
if this is expected to have less than < 10000~ it might be better to use a btree too here